lib/puppet/environments.rb

```diff
@@ -398,14 +398,15 @@ def get_entry(name)
       # This strategy favors smaller memory footprint over environment
       # retrieval time.
       clear_all_expired
+      name = name.to_sym
       entry = @cache[name]
       if entry
         Puppet.debug {"Found in cache #{name.inspect} #{entry.label}"}
         # found in cache
         entry.touch
-      elsif (result = @loader.get(name))
+      elsif (entry = @loader.get(name))
         # environment loaded, cache it
-        entry = entry(result)
+        entry = entry(entry)
```

Comment:
nit, `@loader.get(name)` returns an environment (or nil), so calling the variable `entry` is a little misleading. Also `result` is misleading, because it's no longer returned (as of a00d1374aff8f848a580182290f31e5ac8f1b9e1). So maybe `env`?

```ruby
elsif (env = @loader.get(name))
  entry = entry(env)
  ...
```